aea/cli/upgrade.py

```diff
@@ -0,0 +1,315 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""Implementation of the 'aea upgrade' subcommand."""
+from collections import defaultdict
+from pathlib import Path
+from typing import Any, Generator, Tuple, cast
+
+import click
+
+from aea.cli.add import add_item
+from aea.cli.registry.add import _get_package_meta
+from aea.cli.remove import remove_item
+from aea.cli.utils.click_utils import PublicIdParameter
+from aea.cli.utils.config import load_item_config
+from aea.cli.utils.context import Context
+from aea.cli.utils.decorators import check_aea_project, clean_after, pass_ctx
+from aea.cli.utils.package_utils import (
+    find_item_locally,
+    get_item_public_id_by_author_name,
+    get_items,
+    is_item_present,
+)
+from aea.configurations.base import AgentConfig, PackageConfiguration, PublicId
+
+
+class ItemRemoveHelper:
+    """Helper to check dependencies on removing component from agent config."""
+
+    def __init__(self, agent_config: AgentConfig) -> None:
+        """Init helper."""
+        self._agent_config = agent_config
+
+    def get_agent_dependencies(self) -> defaultdict:
```

Comment:
```suggestion
    def get_agent_reverse_dependency_tree(self) -> defaultdict:
```

Reply:
it's not a tree in common meaning, think like `List[Tuple[PublicID, Set(PublicIDs)]]`